src/Go/boardAnalysis/goAI.ts

```diff
@@ -57,6 +56,11 @@ export function makeAIMove(boardState: BoardState, useOfflineCycles = true): Pro
           return play;
         }
 
+        if (currentMoveCount !== Go.currentGame.previousBoards.length) {
+          console.error("AI move attempted, but the board state has changed.");
+          return play;
+        }
+
         // Handle AI making a move
         await waitCycle(useOfflineCycles);
```

Comment:
I think there is a race where you could do the check and then the state could change over the duration of this `waitCycle`, it should be moved to after here.